extensions/amp-analytics/0.1/events.js

```diff
@@ -687,18 +687,17 @@ export class VisibilityTracker extends EventTracker {
 
     user().assert(SUPPORT_WAITFOR_TRACKERS[waitForSpec] !== undefined,
         'waitFor value %s not supported', waitForSpec);
-
     if (!SUPPORT_WAITFOR_TRACKERS[waitForSpec]) {
-      // waitFor NONE, wait for nothing
       return null;
     }
 
-    if (!this.waitForTrackers_[waitForSpec]) {
-      this.waitForTrackers_[waitForSpec] =
-        new SUPPORT_WAITFOR_TRACKERS[waitForSpec](this.root);
+    let waitForTracker = this.waitForTrackers_[waitForSpec];
+    if (!waitForTracker) {
+      waitForTracker = this.root.getTrackerForOptions(
```

Comment:
`root.getTrackerForOptions()` can return a null, it's safer to check the value before calling `waitForTracker.getElementSignal`. 
I know this should never happen here because we've checked `SUPPORT_WAITFOR_TRACKERS[waitForSpec]` before. But still. Maybe it would be better to move around the check.